### ISSUE TITLE
Support running in rootless mode 

### ISSUE THREAD

**Issue by soenkeliebau** [stackabletech/secret-operator#243]
Some users have expressed a concern that the secret operator needs to run as root, afaik we only need this for the ramdisk that the secrets are stored on.

We want to implement a mode that does not need elevated privileges even if that means that secrets are stored on disk.

**Comment by sbernauer** [comment]
FYI, justed test this on OpenShift
We assigned the secret operator daemonset the scc hostaccess and removed the spec.<container>.securityContext (which was set to user: 0 and privileged: true). We got the following error message
```
# daemonsets.apps "secret-operator-daemonset" was not valid:
# * spec.template.spec.containers[0].volumeMounts.mountPropagation: Forbidden: Bidirectional mount propagation is available only to privileged containers
# * spec.template.spec.containers[0].volumeMounts.mountPropagation: Forbidden: Bidirectional mount propagation is available only to privileged containers
```
TLDR: It seems we need to run as privileged container to use the bidirectional mount
